app/models/post.rb

```diff
@@ -1,12 +1,15 @@
 # Top-level documentation comment for Post module
 class Post < ApplicationRecord
-  # belongs_to :author, class_name: 'User'
-  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id', counter_cache: true
   has_many :comments
   has_many :likes
 
-  def update_user_posts_counter
+  after_create :update_user_counters
+  after_destroy :update_user_counters
+
+  def update_user_counters
     author.update(posts_counter: author.posts.count)
+    update(comments_counter: comments.count, likes_counter: likes.count)
   end
 
```

Comment:
- You have done great so far :+1: But there are still some issues and requirements you need to meet before moving to the next milestone. :+1: :100: 

- [ ] While trying to fix the counter issue from the previous reviewer, it seems you created another issue. When I tried creating a post from the terminal after creating a user, I got this error in the console as shown in the screenshot below, The error seems to reference and unknown attribute. I'll suggest you look into this. :+1: :+1: 

- - Error screenshot:
![Selection_078](https://github.com/yayaismaail/blog/assets/107632780/98d69a98-9032-47f8-9120-2a0ef8f06879)

- [ ] According to the requirement, the `update_post_user_counter` is meant to only update the author's post count and nothing more. I'll suggest you remove these `update(comments_counter: comments.count, likes_counter: likes.count)` as it is not part of what this method is meant to do. :+1: :+1: :wave:

Reply:
Hi @V-Blaze and @Reem-lab, please I need your assistance